resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh

```diff
@@ -65,6 +65,7 @@ case "$1" in
     shift 1
     CMD=(
       "$SPARK_HOME/bin/spark-submit"
+      --driver-java-options '-XX:OnOutOfMemoryError="kill -9 %p"'
```

Comment:
We are in client mode so best to use this arg. Also this is going to be parsed by the `SparkSubmitCommandBuilder` later on.

Reply:
Makes sense - the parent process (think it's `bin/spark-class`) will eventually route this as a JVM argument to the child driver process.

Reply:
Shouldn't we be adding this for executors also?

Reply:
@mccheah afaik executors have a handler for uncaught exceptions and afaik they will handle this and exit:
https://github.com/apache/spark/blob/453cbf3dd8df5ec4da844c93eb6000610b551541/core/src/main/scala/org/apache/spark/executor/Executor.scala#L62  & https://github.com/apache/spark/blob/453cbf3dd8df5ec4da844c93eb6000610b551541/core/src/main/scala/org/apache/spark/executor/Executor.scala#L88